generator/src/main/java/org/apache/kafka/message/MessageSpec.java

```diff
@@ -70,6 +74,12 @@ public MessageSpec(@JsonProperty("name") String name,
                 "messages with type `request`");
         }
         this.listeners = listeners;
+
+        if (latestVersionUnstable && type != MessageSpecType.REQUEST) {
```

Comment:
Makes sense to restrict this for now, but it would be nice to make it more general in the future. I can imagine it being useful while experimenting with metadata record changes.

Reply:
Yeah, I agree.